[2024-12-09T01:54:10.610+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-09T01:54:10.625+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [queued]>
[2024-12-09T01:54:10.631+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [queued]>
[2024-12-09T01:54:10.632+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-09T01:54:10.642+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-08 00:00:00+00:00
[2024-12-09T01:54:10.648+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=659) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-09T01:54:10.648+0000] {standard_task_runner.py:72} INFO - Started process 661 to run task
[2024-12-09T01:54:10.649+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'scheduled__2024-12-08T00:00:00+00:00', '--job-id', '284', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmpe3z9p2mx']
[2024-12-09T01:54:10.651+0000] {standard_task_runner.py:105} INFO - Job 284: Subtask python_job
[2024-12-09T01:54:10.696+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [running]> on host 2a892a3469ec
[2024-12-09T01:54:10.767+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-08T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-08T00:00:00+00:00'
[2024-12-09T01:54:10.768+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-09T01:54:10.788+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-09T01:54:10.789+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-09T01:54:15.429+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO SparkContext: Running Spark version 3.5.3
[2024-12-09T01:54:15.437+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-09T01:54:15.438+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO SparkContext: Java version 17.0.13
[2024-12-09T01:54:15.578+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-09T01:54:15.828+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceUtils: ==============================================================
[2024-12-09T01:54:15.829+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-09T01:54:15.831+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceUtils: ==============================================================
[2024-12-09T01:54:15.832+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO SparkContext: Submitted application: CSVReadExample
[2024-12-09T01:54:15.873+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-09T01:54:15.894+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceProfile: Limiting resource is cpu
[2024-12-09T01:54:15.895+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:15 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-09T01:54:16.044+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SecurityManager: Changing view acls to: ***
[2024-12-09T01:54:16.045+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SecurityManager: Changing modify acls to: ***
[2024-12-09T01:54:16.046+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SecurityManager: Changing view acls groups to:
[2024-12-09T01:54:16.050+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SecurityManager: Changing modify acls groups to:
[2024-12-09T01:54:16.052+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-09T01:54:16.578+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO Utils: Successfully started service 'sparkDriver' on port 38989.
[2024-12-09T01:54:16.674+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SparkEnv: Registering MapOutputTracker
[2024-12-09T01:54:16.774+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-09T01:54:16.804+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-09T01:54:16.805+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-09T01:54:16.816+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-09T01:54:16.864+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-182b45b7-9287-4fbf-ab09-c8874b512a96
[2024-12-09T01:54:16.902+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-09T01:54:16.941+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:16 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-09T01:54:17.212+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-09T01:54:17.341+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-12-09T01:54:17.556+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-09T01:54:17.625+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 33 ms (0 ms spent in bootstraps)
[2024-12-09T01:54:17.773+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241209015417-0003
[2024-12-09T01:54:17.783+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38467.
[2024-12-09T01:54:17.784+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO NettyBlockTransferService: Server created on 2a892a3469ec:38467
[2024-12-09T01:54:17.785+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-09T01:54:17.793+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 38467, None)
[2024-12-09T01:54:17.797+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:38467 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 38467, None)
[2024-12-09T01:54:17.800+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 38467, None)
[2024-12-09T01:54:17.801+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:17 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 38467, None)
[2024-12-09T01:54:18.042+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:18 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-09T01:54:18.411+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:18 INFO SparkContext: Added file /tmp/test1234.csv at spark://2a892a3469ec:38989/files/test1234.csv with timestamp 1733709258410
[2024-12-09T01:54:18.413+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:18 INFO Utils: Copying /tmp/test1234.csv to /tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv
[2024-12-09T01:54:18.438+0000] {spark_submit.py:634} INFO - Resolved file path by Spark: /tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv
[2024-12-09T01:54:18.447+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:18 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-12-09T01:54:18.449+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:18 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-12-09T01:54:20.159+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:20 INFO InMemoryFileIndex: It took 53 ms to list leaf files for 1 paths.
[2024-12-09T01:54:20.260+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:20 INFO InMemoryFileIndex: It took 3 ms to list leaf files for 1 paths.
[2024-12-09T01:54:22.310+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:22 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241209015417-0003/0 on worker-20241208214935-172.18.0.7-42783 (172.18.0.7:42783) with 2 core(s)
[2024-12-09T01:54:22.321+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:22 INFO StandaloneSchedulerBackend: Granted executor ID app-20241209015417-0003/0 on hostPort 172.18.0.7:42783 with 2 core(s), 1024.0 MiB RAM
[2024-12-09T01:54:22.434+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:22 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241209015417-0003/0 is now RUNNING
[2024-12-09T01:54:23.530+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:23 INFO FileSourceStrategy: Pushed Filters:
[2024-12-09T01:54:23.534+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:23 INFO FileSourceStrategy: Post-Scan Filters: (length(trim(value#0, None)) > 0)
[2024-12-09T01:54:24.184+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO CodeGenerator: Code generated in 209.402 ms
[2024-12-09T01:54:24.251+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 199.6 KiB, free 434.2 MiB)
[2024-12-09T01:54:24.316+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 34.4 KiB, free 434.2 MiB)
[2024-12-09T01:54:24.323+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 2a892a3469ec:38467 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-09T01:54:24.326+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO SparkContext: Created broadcast 0 from csv at NativeMethodAccessorImpl.java:0
[2024-12-09T01:54:24.340+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO FileSourceScanExec: Planning scan with bin packing, max size: 4194304 bytes, open cost is considered as scanning 4194304 bytes.
[2024-12-09T01:54:24.446+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO SparkContext: Starting job: csv at NativeMethodAccessorImpl.java:0
[2024-12-09T01:54:24.472+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Got job 0 (csv at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-12-09T01:54:24.473+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Final stage: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0)
[2024-12-09T01:54:24.474+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Parents of final stage: List()
[2024-12-09T01:54:24.474+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Missing parents: List()
[2024-12-09T01:54:24.480+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-12-09T01:54:24.569+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 13.5 KiB, free 434.2 MiB)
[2024-12-09T01:54:24.578+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 6.4 KiB, free 434.2 MiB)
[2024-12-09T01:54:24.580+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 2a892a3469ec:38467 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-09T01:54:24.582+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1585
[2024-12-09T01:54:24.606+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-12-09T01:54:24.607+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:24 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-12-09T01:54:25.319+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:25 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.7:51866) with ID 0,  ResourceProfileId 0
[2024-12-09T01:54:25.363+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:25 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.7:43627 with 434.4 MiB RAM, BlockManagerId(0, 172.18.0.7, 43627, None)
[2024-12-09T01:54:25.404+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:25 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-09T01:54:25.628+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:25 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 172.18.0.7:43627 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-09T01:54:26.215+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.18.0.7:43627 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-09T01:54:26.562+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.562+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-09T01:54:26.563+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-09T01:54:26.564+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-09T01:54:26.564+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-09T01:54:26.564+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-09T01:54:26.564+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-09T01:54:26.564+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-09T01:54:26.565+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-09T01:54:26.566+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-09T01:54:26.566+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-09T01:54:26.566+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-09T01:54:26.567+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-09T01:54:26.568+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-09T01:54:26.568+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-09T01:54:26.568+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-09T01:54:26.568+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T01:54:26.569+0000] {spark_submit.py:634} INFO - 
[2024-12-09T01:54:26.569+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Starting task 0.1 in stage 0.0 (TID 1) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-09T01:54:26.596+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Lost task 0.1 in stage 0.0 (TID 1) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.596+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 1]
[2024-12-09T01:54:26.597+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Starting task 0.2 in stage 0.0 (TID 2) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-09T01:54:26.613+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Lost task 0.2 in stage 0.0 (TID 2) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.614+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 2]
[2024-12-09T01:54:26.614+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Starting task 0.3 in stage 0.0 (TID 3) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-09T01:54:26.633+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSetManager: Lost task 0.3 in stage 0.0 (TID 3) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.633+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 3]
[2024-12-09T01:54:26.634+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
[2024-12-09T01:54:26.635+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool
[2024-12-09T01:54:26.636+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSchedulerImpl: Cancelling stage 0
[2024-12-09T01:54:26.637+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage cancelled: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.637+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-09T01:54:26.637+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-09T01:54:26.637+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-09T01:54:26.637+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-09T01:54:26.638+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-09T01:54:26.638+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-09T01:54:26.638+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-09T01:54:26.638+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-09T01:54:26.638+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-09T01:54:26.639+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-09T01:54:26.639+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-09T01:54:26.639+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-09T01:54:26.639+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-09T01:54:26.639+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-09T01:54:26.640+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-09T01:54:26.640+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-09T01:54:26.640+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-09T01:54:26.640+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-09T01:54:26.640+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-09T01:54:26.641+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-09T01:54:26.641+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-09T01:54:26.641+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-09T01:54:26.641+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-09T01:54:26.641+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-09T01:54:26.642+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-09T01:54:26.642+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T01:54:26.642+0000] {spark_submit.py:634} INFO - 
[2024-12-09T01:54:26.642+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-09T01:54:26.642+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO DAGScheduler: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0) failed in 2.148 s due to Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-09T01:54:26.643+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-09T01:54:26.644+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-09T01:54:26.644+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-09T01:54:26.644+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-09T01:54:26.644+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-09T01:54:26.644+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-09T01:54:26.645+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-09T01:54:26.646+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-09T01:54:26.647+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-09T01:54:26.647+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T01:54:26.647+0000] {spark_submit.py:634} INFO - 
[2024-12-09T01:54:26.647+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-09T01:54:26.647+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO DAGScheduler: Job 0 failed: csv at NativeMethodAccessorImpl.java:0, took 2.197051 s
[2024-12-09T01:54:26.799+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-09T01:54:26.799+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 37, in <module>
[2024-12-09T01:54:26.800+0000] {spark_submit.py:634} INFO - df = spark.read.csv(path=myfile, header=True, inferSchema=True)
[2024-12-09T01:54:26.800+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-09T01:54:26.801+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 740, in csv
[2024-12-09T01:54:26.801+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-12-09T01:54:26.801+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-12-09T01:54:26.802+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-12-09T01:54:26.843+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o27.csv.
[2024-12-09T01:54:26.846+0000] {spark_submit.py:634} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.847+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-09T01:54:26.848+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-09T01:54:26.849+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-09T01:54:26.850+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-09T01:54:26.851+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-09T01:54:26.853+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-09T01:54:26.853+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-09T01:54:26.853+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-09T01:54:26.853+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-09T01:54:26.853+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-09T01:54:26.854+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-09T01:54:26.855+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-09T01:54:26.855+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-09T01:54:26.855+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-09T01:54:26.855+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-09T01:54:26.855+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-09T01:54:26.856+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-09T01:54:26.857+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-09T01:54:26.857+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-09T01:54:26.857+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T01:54:26.857+0000] {spark_submit.py:634} INFO - 
[2024-12-09T01:54:26.857+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-09T01:54:26.858+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-12-09T01:54:26.858+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-12-09T01:54:26.858+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-12-09T01:54:26.859+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-12-09T01:54:26.860+0000] {spark_submit.py:634} INFO - at scala.Option.foreach(Option.scala:407)
[2024-12-09T01:54:26.860+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-12-09T01:54:26.860+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-12-09T01:54:26.861+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-12-09T01:54:26.862+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-12-09T01:54:26.863+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:530)
[2024-12-09T01:54:26.864+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:483)
[2024-12-09T01:54:26.864+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:61)
[2024-12-09T01:54:26.864+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.collectFromPlan(Dataset.scala:4333)
[2024-12-09T01:54:26.864+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$head$1(Dataset.scala:3316)
[2024-12-09T01:54:26.864+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4323)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4321)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-12-09T01:54:26.867+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.head(Dataset.scala:3316)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.take(Dataset.scala:3539)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.TextInputCSVDataSource$.infer(CSVDataSource.scala:111)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVDataSource.inferSchema(CSVDataSource.scala:64)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVFileFormat.inferSchema(CSVFileFormat.scala:62)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.$anonfun$getOrInferFileFormatSchema$11(DataSource.scala:208)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at scala.Option.orElse(Option.scala:447)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.getOrInferFileFormatSchema(DataSource.scala:205)
[2024-12-09T01:54:26.868+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:407)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.csv(DataFrameReader.scala:538)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-09T01:54:26.869+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-09T01:54:26.872+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-09T01:54:26.873+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T01:54:26.876+0000] {spark_submit.py:634} INFO - Caused by: org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/userFiles-163bde25-974c-40b2-8e9a-9ac03f1e8952/test1234.csv does not exist
[2024-12-09T01:54:26.877+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-09T01:54:26.877+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-09T01:54:26.877+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-09T01:54:26.877+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-09T01:54:26.878+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-09T01:54:26.880+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-09T01:54:26.881+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - ... 1 more
[2024-12-09T01:54:26.882+0000] {spark_submit.py:634} INFO - 
[2024-12-09T01:54:26.915+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-09T01:54:26.916+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-09T01:54:26.924+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4040
[2024-12-09T01:54:26.927+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-09T01:54:26.928+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-09T01:54:26.946+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-09T01:54:26.966+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO MemoryStore: MemoryStore cleared
[2024-12-09T01:54:26.966+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO BlockManager: BlockManager stopped
[2024-12-09T01:54:26.970+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-09T01:54:26.972+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-09T01:54:26.981+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO SparkContext: Successfully stopped SparkContext
[2024-12-09T01:54:26.981+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO ShutdownHookManager: Shutdown hook called
[2024-12-09T01:54:26.981+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO ShutdownHookManager: Deleting directory /tmp/spark-b955af6a-6050-4b80-a567-ada259a35262/pyspark-fd9cd8b8-1adf-43bf-a4bd-c7d1d9061184
[2024-12-09T01:54:26.985+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO ShutdownHookManager: Deleting directory /tmp/spark-1b480aea-b6b2-4ccb-b0f9-5cf297d39ce0
[2024-12-09T01:54:26.988+0000] {spark_submit.py:634} INFO - 24/12/09 01:54:26 INFO ShutdownHookManager: Deleting directory /tmp/spark-b955af6a-6050-4b80-a567-ada259a35262
[2024-12-09T01:54:27.036+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T01:54:27.047+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=scheduled__2024-12-08T00:00:00+00:00, execution_date=20241208T000000, start_date=20241209T015410, end_date=20241209T015427
[2024-12-09T01:54:27.060+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-09T01:54:27.061+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 284 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 661)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T01:54:27.091+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-09T01:54:27.115+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-09T01:54:27.118+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-12-09T19:43:40.304+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-09T19:43:40.333+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [queued]>
[2024-12-09T19:43:40.346+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [queued]>
[2024-12-09T19:43:40.347+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-09T19:43:40.369+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-08 00:00:00+00:00
[2024-12-09T19:43:40.382+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=197) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-09T19:43:40.380+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'scheduled__2024-12-08T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmpnj5op0pt']
[2024-12-09T19:43:40.385+0000] {standard_task_runner.py:72} INFO - Started process 201 to run task
[2024-12-09T19:43:40.385+0000] {standard_task_runner.py:105} INFO - Job 4: Subtask python_job
[2024-12-09T19:43:40.475+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job scheduled__2024-12-08T00:00:00+00:00 [running]> on host 0f591f475d73
[2024-12-09T19:43:40.600+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Harsha Vardhan' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-08T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-08T00:00:00+00:00'
[2024-12-09T19:43:40.601+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-09T19:43:40.891+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-09T19:43:40.897+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master :7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-09T19:43:49.401+0000] {spark_submit.py:634} INFO - Exception in thread "main" org.apache.spark.SparkException: Master must either be yarn or start with spark, mesos, k8s, or local
[2024-12-09T19:43:49.429+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.error(SparkSubmit.scala:1047)
[2024-12-09T19:43:49.430+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:256)
[2024-12-09T19:43:49.435+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:964)
[2024-12-09T19:43:49.436+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:194)
[2024-12-09T19:43:49.437+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:217)
[2024-12-09T19:43:49.472+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
[2024-12-09T19:43:49.475+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1120)
[2024-12-09T19:43:49.486+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1129)
[2024-12-09T19:43:49.487+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-12-09T19:43:49.748+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master :7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T19:43:49.783+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=scheduled__2024-12-08T00:00:00+00:00, execution_date=20241208T000000, start_date=20241209T194340, end_date=20241209T194349
[2024-12-09T19:43:49.835+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-09T19:43:49.837+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 4 for task python_job (Cannot execute: spark-submit --master :7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 201)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master :7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T19:43:49.873+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-09T19:43:49.950+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
